### PR TITLE
drivers: qspi: added operation timeout

### PR DESCRIPTION
**Pull request by hlord2000**
Added devicetree assignment of qspi timeout.
Per nrfx v3.2 addition of qspi timeout in config
struct.

Signed-off-by: Kelly Helmut Lord <kellyhlord@gmail.com>

### CODE REVIEW

File: dts/bindings/flash_controller/nordic,nrf-qspi.yaml [hunk 1]
```diff
@@ -39,3 +39,8 @@ properties:
 
   pinctrl-0:
     required: true
+
+  timeout:
+    type: int
+    default: 500
+    description: Timeout for QSPI operations in ms.
```
Comment (gmarull) [inline dts/bindings/flash_controller/nordic,nrf-qspi.yaml:46]: First, this seems to be a software setting, ie, does not belong to DT.

But it also has DT issues:
- Not ussing suffix (-ms)
- Default not justified (random value here)

**Review by gmarull — CHANGES_REQUESTED** [review]

File: drivers/flash/Kconfig.nordic_qspi_nor [hunk 2]
```diff
@@ -50,4 +50,13 @@ config NORDIC_QSPI_NOR_XIP
...
 	  This will also disable power management for the QSPI NOR flash chip.
 
+config NORDIC_QSPI_NOR_TIMEOUT_MS
+	int "Timeout for QSPI operations (ms)"
+	default 500
+	help
+	  The QSPI peripheral operation timeout in milliseconds.
+	  Primarily intended for long running operations such as
+	  flash mass erase. 500 ms is the default value as
+	  set in the nrfx QSPI driver.
```
Comment (gmarull) [inline drivers/flash/Kconfig.nordic_qspi_nor:60]: This is not a good explanation, nrfx should not be referenced here (Zephyr users may not care about nrfx at all). Just give a brief explanation of why 500ms is a reasonable value.

Comment (hlord2000) [reply drivers/flash/Kconfig.nordic_qspi_nor]: I have updated this to reflect typical sector erase time specs

**Review by gmarull — APPROVED** [review]

**Review by de-nordic — APPROVED** [review]

**Review by de-nordic — APPROVED** [review]
Can you add commit with Release Notes entry documenting the addition of Kconfig?

**Pull request merged**